framework/applications/noviusos_menu/classes/model/menu.model.php

```diff
@@ -175,4 +175,77 @@ private static function addPagesItems($context, $idParent, $depth)
         }
         return $items;
     }
+
+    /**
+     * @param $targetContext : the context target wanted for the duplicated menu
+     * @throws \Exception
+     */
+    public function duplicate($targetContext)
+    {
+        $clone = clone $this;
+        $try = 1;
+        do {
+            try {
+                $title_append = __(' (copy)');
+                $clone->menu_title = $this->title_item().$title_append;
+                $clone->menu_context = $targetContext;
+                $clone->menu_context_common_id = $clone->menu_id;
+
+                if ($clone->save()) {
+                    static::duplicateMenuItems($this, $clone);
```

Comment:
Why do you need statically call duplicateMenuItems and duplicateMenuItems in this context?